crates/deltalake-test/src/utils.rs

```diff
@@ -183,88 +144,6 @@ impl IntegrationContext {
     }
 }
 
-impl Drop for IntegrationContext {
-    fn drop(&mut self) {
-        match self.integration {
-            StorageIntegration::Amazon => {
-                s3_cli::delete_bucket(self.root_uri()).unwrap();
-                s3_cli::delete_lock_table().unwrap();
-            }
-            StorageIntegration::Microsoft => {
-                az_cli::delete_container(&self.bucket).unwrap();
-            }
-            StorageIntegration::Google => {
-                gs_cli::delete_bucket(&self.bucket).unwrap();
-            }
-            StorageIntegration::Onelake => (),
-            StorageIntegration::OnelakeAbfs => (),
-            StorageIntegration::Local => (),
-            StorageIntegration::Hdfs => {
-                hdfs_cli::delete_dir(&self.bucket).unwrap();
-            }
-        };
-        self.restore_env();
-    }
-}
-
-/// Kinds of storage integration
-pub enum StorageIntegration {
-    Amazon,
-    Microsoft,
-    Onelake,
-    Google,
-    Local,
-    Hdfs,
-    OnelakeAbfs,
-}
-
-impl StorageIntegration {
```

Comment:
Are these things not used anymore for the other cloud stores?

Reply:
For the other cloud stores they should be implementing the trait. I don't actually know why our integration tests were implemented the way they that they were, basically with everything in kind of hard-coded lists with feature flags.

The model used in `deltalake-aws/tests/common.rs` is what I would expect to see more of for the other cloud provider traits